feat(ColorScaleBlock): adapt resizing and drag and drop

celine-s opened the pull request.

dominiquemb reviewed (changes requested):
@celine-s I tested this locally and noticed a few things need attention.

1) The grey placeholders are not rendering when the block is empty: 
<img width="907" alt="Screenshot 2023-01-19 at 13 44 09" src="https://user-images.githubusercontent.com/4213299/213447004-e8aad9a3-e275-4d0d-85f8-c546334d6336.png">


2) When I first add a color, the block appears empty with a border around it:
<img width="909" alt="Screenshot 2023-01-19 at 13 44 18" src="https://user-images.githubusercontent.com/4213299/213447057-fdd0395a-9886-4438-9a7a-7fc4484fdc7a.png">

3) When I add a 2nd color, the 1st color renders, however it only takes up the minimum width. All colors, unless they have been resized manually, should always take up an even amount of space in the block. If there are 2 colors, they should each have 50% width, and if there are 3, they should take up 33% width, and so on.
<img width="911" alt="Screenshot 2023-01-19 at 13 44 31" src="https://user-images.githubusercontent.com/4213299/213447297-a27fc866-6919-42c5-aedc-2a61568d7ec2.png">

4) Despite the fact that I did not resize the 2nd color, when I add a 3rd color, the 2nd color seems to take up an unexpected amount of space.
<img width="913" alt="Screenshot 2023-01-19 at 13 45 16" src="https://user-images.githubusercontent.com/4213299/213447475-6c446fc7-fc4e-4f73-b78b-744a97402e7c.png">

celine-s commented:
> 

Uh, good catch. Should be fixed. Can you check again?

dominiquemb reviewed (changes requested):
@celine-s A few issues still remain, but it mostly looks good.

1) The delete button should be the same color as the settings buttons on the right of the block.
<img width="241" alt="Screenshot 2023-01-19 at 15 59 08" src="https://user-images.githubusercontent.com/4213299/213478111-d11c4a77-c71b-4697-9b98-a5aa0074fb99.png">

2. When dragging, the color square has a white line at the bottom which shouldn't be there.
<img width="1728" alt="Screenshot 2023-01-19 at 16 01 34" src="https://user-images.githubusercontent.com/4213299/213478245-9527096d-5b1f-4bff-8d08-ac551b48e186.png">

3. When there is no room to add a new color, the 'Add Color' button should be disabled. It should also be re-enabled whenever there is extra room for a new color.
<img width="136" alt="Screenshot 2023-01-19 at 16 00 45" src="https://user-images.githubusercontent.com/4213299/213478320-5ad2d3ad-0e43-4d12-8396-cf4f499b4abb.png">

4. When hovering over the resize handle, the cursor should change to indicate that dragging is possible.
<img width="1728" alt="Screenshot 2023-01-19 at 16 01 26" src="https://user-images.githubusercontent.com/4213299/213478572-8e70c435-47db-4267-9f04-154d1b9e8f9c.png">

celine-s commented:
@dominiquemb 

Thanks for checking again!!

1. nice catch. Fixed.
2. True that. I did not find a fast fix therefore I added it to the ticket [here](https://app.clickup.com/t/86775v2qg) that it is still remaining and we can tackle it later.
3. At the moment it will always take from all the colors percentage wise space. As long as this is possible it would be possible to add colors. Is the wanted behaviour to just check the last color?
4. I can not reproduce it. I moved the handle a bit to the left maybe that was the issue. Or do you still have it?
5. shouldn't be the case. I think you have some old colors in it. Could you delete the block and add a new one. Do you then still have it?

dominiquemb commented:
> @dominiquemb Thanks for checking again!!
> 
> According to your points
> 
> 1. nice catch. Fixed.
> 2. True that. I did not find a fast fix therefore I added it to the ticket [here](https://app.clickup.com/t/86775v2qg) that it is still remaining and we can tackle it later.
> 3. At the moment it will always take from all the colors percentage wise space. As long as this is possible it would be possible to add colors. Is the wanted behaviour to just check the last color?
> 4. I can not reproduce it. I moved the handle a bit to the left maybe that was the issue. Or do you still have it?
> 5. shouldn't be the case. I think you have some old colors in it. Could you delete the block and add a new one. Do you then still have it?

3. When a color has been resized, its width should not be adjusted automatically whenever a new color is added. As a result, sometimes there may be one big color, and one small color at the end, resulting in there not being enough space to add any new colors. In this case, the 'add color' button needs to be disabled.  I can describe the desired behavior to you on a call, if you want.

4. When I hover over the drag handle, I still cannot see a 'resize' cursor. 
5. Same thing, I still cannot see the grey placeholders yet.

celine-s commented on packages/color-scale-block/src/ColorScaleBlock.tsx:
```diff
@@ -99,368 +79,152 @@ export const ColorScaleBlock: FC<BlockProps> = ({ appBridge }) => {
+                                                }}
+                                                className="tw-absolute group-hover:tw-flex -tw-ml-1 tw-w-2 tw-flex tw-h-full tw-items-center tw-justify-center tw-opacity-0 hover:tw-opacity-100"
+                                                onMouseDown={(e) => {
+                                                    const startX = e.clientX;
+                                                    const startWidth = displayableItems[idx].width;
+                                                    const nextWidth = displayableItems[idx + 1].width;
+
+                                                    const onMouseMove = (e: MouseEvent) => {
+                                                        const diff = e.clientX - startX;
+                                                        const percentage =
+                                                            totalWidth /
+                                                            (colorScaleBlockRef.current?.clientWidth ?? 800);
+                                                        const newWidth = startWidth + diff * percentage;
+                                                        const newNextWidth = nextWidth - diff * percentage;
+                                                        if (
+                                                            newWidth > MINIMUM_COLOR_WIDTH &&
+                                                            newNextWidth > MINIMUM_COLOR_WIDTH
+                                                        ) {
+                                                            setDisplayableItems((prev) => {
+                                                                const newColors = [...prev];
+                                                                newColors[idx].width = newWidth;
+                                                                newColors[idx + 1].width = newNextWidth;
+                                                                return newColors;
+                                                            });
+                                                        }
+                                                    };
+                                                    const onMouseUp = () => {
+                                                        window.removeEventListener('mousemove', onMouseMove);
+                                                        window.removeEventListener('mouseup', onMouseUp);
+                                                        setBlockSettings({ colorInput: displayableItems });
+                                                    };
+                                                    window.addEventListener('mousemove', onMouseMove);
+                                                    window.addEventListener('mouseup', onMouseUp);
+                                                }}
+                                            >
+                                                <div
+                                                    data-test-id="drag-handle"
+                                                    className="tw-bg-black-20 tw-rounded-full tw-w-2 tw-h-6"
+                                                />
+                                            </div>
```
These lines are more or less the key of this open PR. For someone who takes it over: this resizing function is straight forward and nice to have for this block. 

cc @ragi96
